solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/ModelQuery.java

```diff
@@ -267,37 +267,34 @@ public FeatureWeight call() throws Exception{
   } // end of call CreateWeightCallable
 
   private void createWeightsParallel(IndexSearcher searcher, boolean needsScores, float boost,
-      List<FeatureWeight > featureWeights, Collection<Feature> features) throws IOException {
+      List<FeatureWeight > featureWeights, Collection<Feature> features) throws RuntimeException {
 
     final SolrQueryRequest req = getRequest();
-
     Executor executor = LTRThreadModule.createWeightScoreExecutor;
-    if  (LTRThreadModule.ltrSemaphore == null ){
-      LTRThreadModule.ltrSemaphore = new Semaphore(LTRThreadModule.getMaxThreads());
-    }
     List<Future<FeatureWeight> > futures = new ArrayList<>(features.size());
     try{
       for (final Feature f : features) {
         CreateWeightCallable callable = new CreateWeightCallable(f, searcher, needsScores, req);
         RunnableFuture<FeatureWeight> runnableFuture = new FutureTask<>(callable);
         querySemaphore.acquire(); // always acquire before the ltrSemaphore is acquired, to guarantee a that the current query is within the limit for max. threads 
         LTRThreadModule.ltrSemaphore.acquire();//may block and/or interrupt
-        
         executor.execute(runnableFuture);//releases semaphore when done
         futures.add(runnableFuture);
       }
       //Loop over futures to get the feature weight objects
       for (final Future<FeatureWeight> future : futures) {
-        featureWeights.add(future.get());
+        featureWeights.add(future.get()); // future.get() will block if the job is still running
       }
     } catch (InterruptedException e) {
       log.info("Error while creating weights in LTR: InterruptedException", e);
+      throw new RuntimeException("Error while creating weights in LTR");
```

Comment:
do we need to catch and rethrow, can we just delete the try/catch and fail instead?